pkg/build/controller/controller_test.go

```diff
@@ -8,22 +8,49 @@ import (
 	kerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
+	builder "github.com/openshift/origin/pkg/build/builder"
 	buildtest "github.com/openshift/origin/pkg/build/controller/test"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
-type okBuildUpdater struct{}
+type okBuildClient struct{}
 
-func (obu *okBuildUpdater) UpdateBuild(namespace string, build *buildapi.Build) (*buildapi.Build, error) {
+func (okc *okBuildClient) UpdateBuild(namespace string, build *buildapi.Build) (*buildapi.Build, error) {
 	return &buildapi.Build{}, nil
 }
 
-type errBuildUpdater struct{}
+// dummy methods to finish implementing the interface
+func (okc *okBuildClient) CreateBuild(namespace string, build *buildapi.Build) (*buildapi.Build, error) {
```

Comment:
I'm ok with well shared interfaces.  I'm not ok with abstracting those interfaces.  The code that uses an interface should define it - it's good to make those interfaces consistent across their usage, but creating a "spider" interface (in util! no less) that creates a web of unrelated connections for that unrelated function creates strictly more confusion.  Each consumer should take *exactly* the interface they use - not a generic interface.